spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/thymeleaf/ThymeleafAutoConfiguration.java

```diff
@@ -259,4 +264,29 @@ public ResourceUrlEncodingFilter resourceUrlEncodingFilter() {
 
 	}
 
+	@Configuration
+	@ConditionalOnWebApplication
+	@ConditionalOnProperty(name = "spring.thymeleaf.enabled", matchIfMissing = true)
+	protected static class ThymeleafRootViewConfiguration extends WebMvcConfigurerAdapter {
+		private final ThymeleafProperties properties;
+		private final ApplicationContext applicationContext;
+
+		public ThymeleafRootViewConfiguration(final ThymeleafProperties properties,
+				final ApplicationContext applicationContext) {
+			this.properties = properties;
+			this.applicationContext = applicationContext;
+		}
+
+		@Override
+		public void addViewControllers(final ViewControllerRegistry registry) {
+			if (StringUtils.hasText(this.properties.getRootView())) {
+				final String rootLocation = this.properties.getPrefix()
+						+ this.properties.getRootView() + this.properties.getSuffix();
+				final TemplateLocation location = new TemplateLocation(rootLocation);
+				if (location.exists(this.applicationContext)) {
+					registry.addViewController("/").setViewName(this.properties.getRootView());
```

Comment:
I changed this as you suggest. WelcomePageHandlerMapping and WelcomeTemplateHandlerMapping are now almost identical. Also, as I add this support to other template engines, they're going to have this almost identical setup as well. Should this be extracted to some other base class? And where should I put it? `org.springframework.boot.autoconfigure.web`?